source/SpinalHDL/Libraries/Pipeline/introduction.rst

```diff
@@ -0,0 +1,762 @@
+
+Introduction
+============
+
+spinal.lib.misc.pipeline provide a pipelining API. The main advantages over manual pipelining are : 
+
+- You don't have to pre define data structure with all the element which goes through each individual stage
+- You can compose the pipeline (as a consequence of the above point)
+- Manual retiming is much easier, as you don't have to handle the registers / arbitration manualy
+- Manage the arbitration by itself
+
+The API is composed of 4 main things : 
+
+- Node : which represent a layer in the pipeline
+- Connector : which allows to connect nodes to each others
+- Builder : which will generate the hardware required for a whole pipeline
+- Stageable : which represent a data thing that you can retrieve Nodes the pipeline 
+
+It is important to understande that Stageable isn't a hardware data instance, but a key to retrieve a data along the pipeline on nodes.
```

Comment:
is PipeSignal a good choice？

Reply:
or NodeStreamPayload？

Reply:
or NodeStreamPayload？

Reply:
PipeSignal is kinda of missleading, because a Signal should be a unique thing, while in our case it can be multiple instance through the pipeline.
NodeStreamPayload is too verbose i would say.